subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -64,8 +64,13 @@ static ZTEST_BMEM enum ztest_status test_status = ZTEST_STATUS_OK;
 
 extern ZTEST_DMEM const struct ztest_arch_api ztest_api;
 
+static void __ztest_show_suite_summary(void);
+
 void end_report(void)
 {
+	if (IS_ENABLED(CONFIG_ZTEST_SUMMARY)) {
+		__ztest_show_suite_summary();
```

Comment:
What it the motivation for moving this call to `end_report`? It looks like this forces you to forward-declare `__ztest_show_suite_summary`.

Reply:
To print the summary only once, instead of each time we want to run all suites. Suppose if `test_main` is overridden and there are tests that need to run before an init procedure, then we need to run the remaining test afterwards. In this scenario the summary is printed twice and leads to confusion as the first summary shows a bunch of skipped tests that are ran after init procedure.  Moving it here gives the final summary printed once.

Reply:
Ack.

As an aside, I don't see `end_report` in any headers. Should it be static?

Reply:
done